### PR TITLE
chore(query-performance): Remove unneeded fields when loading a dashboard

### PR DESCRIPTION
**Pull request by macobo**
## Problem

Dashboards load a while. This is due to bad code.

## Changes

Remove _some_ of the low-hanging fruit causing extra reads.

### CODE REVIEW

File: posthog/api/test/dashboards/test_dashboard.py [hunk 1]
```diff
@@ -428,10 +428,8 @@ def test_dashboard_insight_tiles_can_be_loaded_correct_context(self):
         response = self.dashboard_api.get_dashboard(dashboard_id)
         self.assertEqual(len(response["tiles"]), 1)
```
[the remark this answers is not in the document]
Comment (pauldambra) [reply posthog/api/test/dashboards/test_dashboard.py:429]: we must have been checking that items and tiles were present

File: posthog/api/test/dashboards/test_dashboard.py [hunk 2]
```diff
@@ -428,10 +428,8 @@ def test_dashboard_insight_tiles_can_be_loaded_correct_context(self):
...
         self.assertEqual(len(response["tiles"]), 1)
-        item_insight = response["tiles"][0]
         tile = response["tiles"][0]
 
-        assert item_insight["filters_hash"] == tile["filters_hash"]
         assert tile["insight"]["id"] == insight_id
```
Comment (pauldambra) [inline posthog/api/test/dashboards/test_dashboard.py:433]: The interesting thing about removing this is it makes it clear how lazy I was with this test 😅 

What we care about here is what filters have been applied to the insights... We were checking the filters hash as a proxy for that. So would need to assert on something like `response["tiles"][0]["insight"]["filters"]["date_from"] == "-14d"`

File: posthog/api/test/test_insight.py [hunk 3]
```diff
@@ -2003,165 +1985,6 @@ def test_admin_user_can_add_an_insight_to_a_restricted_dashboard(self) -> None:
         )
         assert response.status_code == status.HTTP_200_OK
 
-    def test_saving_an_insight_with_new_filters_updates_the_dashboard_tile(
```
Comment (pauldambra) [inline posthog/api/test/test_insight.py:2006]: I guess these are covered in the `InsightCachingState` tests?

File: posthog/models/dashboard_tile.py [hunk 4]
```diff
@@ -107,37 +105,6 @@ def dashboard_queryset(queryset: QuerySet) -> QuerySet:
         )
 
 
-@mutable_receiver(post_save, sender=Insight)
```
Comment (pauldambra) [inline posthog/models/dashboard_tile.py:110]: I assume you've done the thinking but how do we know these are safe to remove?

If I'm editing and moving backwards and forwards between states aren't these what let me hit the cache?

File: posthog/models/insight.py [hunk 5]
```diff
@@ -174,18 +172,6 @@ class Meta:
     last_viewed_at: models.DateTimeField = models.DateTimeField()
 
 
-@mutable_receiver(pre_save, sender=Insight)
```
Comment (pauldambra) [inline posthog/models/insight.py:177]: as above

**Review by pauldambra — COMMENTED** [review]
Generally speaking fine to remove those API fields...

A few questions on some of the model/api knock-on changes though

File: posthog/models/dashboard_tile.py (hunk 4 above)
Comment (macobo) [reply posthog/models/dashboard_tile.py:110]: Not sure if you mean this, but we don't read Insight/DashboardTile.filters_hash any more. InsightCachingState has its own signal handling in posthog/models/insight_caching_state.py

File: posthog/api/test/test_insight.py (hunk 3 above)
Comment (macobo) [reply posthog/api/test/test_insight.py:2006]: Yes!

File: posthog/models/dashboard_tile.py (hunk 4 above)
Comment (pauldambra) [reply posthog/models/dashboard_tile.py:110]: I'd missed how much tidier that all is now! 🙌

**Review by pauldambra — APPROVED** [review]
One non-blocking thought on one of the tests otherwise 🚢

File: posthog/api/test/dashboards/test_dashboard.py (hunk 2 above)
Comment (macobo) [reply posthog/api/test/dashboards/test_dashboard.py:432]: Thank you, will update!

**Pull request merged**